invoices/invoiceregistry.go

```diff
@@ -419,6 +430,55 @@ func (i *InvoiceRegistry) LookupInvoice(rHash lntypes.Hash) (channeldb.Invoice,
 	return i.cdb.LookupInvoice(rHash)
 }
 
+// processKeySend just-in-time inserts an invoice if this htlc is a key send
+// htlc.
+func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx,
+	hash lntypes.Hash) error {
+
+	preimageSlice, ok := ctx.customRecords[KeySendRecord]
+	if !ok {
+		return nil
+	}
+	// Cancel htlc is preimage is invalid.
+	preimage, err := lntypes.MakePreimage(preimageSlice)
+	if err != nil || preimage.Hash() != hash {
+		return errors.New("invalid key send preimage")
+	}
+
+	// Don't accept zero preimages as those have a special meaning in our
+	// database for hodl invoices.
+	if preimage == channeldb.UnknownPreimage {
+		return errors.New("invalid key send preimage")
+	}
+
+	// Create placeholder invoice.
+	invoiceFeatures := lnwire.NewFeatureVector(
+		lnwire.NewRawFeatureVector(), lnwire.Features,
```

Comment:
See comment about setting a new feature bit here to indicate that this is actually a `keysend` payment. Alternatively, we could just add a new payment type field which stays consistent regardless of the underlying `keysend` implementation (AMP, or custom record, etc).

Reply:
Added a commit that adds this rpc field

Reply:
no need to instantiate the raw feature vector, nil will suffice.

Reply:
instantiated with nil